src/components/Chat/Row.js

```diff
@@ -0,0 +1,189 @@
+  Text,
+  View,
+} from 'react-native';
+import { Actions } from 'react-native-router-flux';
+import moment from 'moment/min/moment.min';
+
+class Row extends Component {
+  constructor(props) {
+    super(props);
+
+    if (props.dataSource.members[0].userId == props._id){
+      this.me = props.dataSource.members[0];
+      this.opponent = props.dataSource.members[1];
+    }else{
+      this.me = props.dataSource.members[1];
+      this.opponent = props.dataSource.members[0];
+    }
+
+    this.goToChat = this.goToChat.bind(this);
+    this.state ={
+      channel : props.dataSource,
+      me :this.me,
+      opponent : this.opponent,
+      lastMessageInfo : props.dataSource.lastMessage,
+      unreadCount : props.dataSource.unreadMessageCount,
+    };
+  }
+
+  componentWillReceiveProps(props) {
+    if (props.dataSource.members[0].userId == props._id){
+      this.me = props.dataSource.members[0];
+      this.opponent = props.dataSource.members[1];
+    }else{
+      this.me = props.dataSource.members[1];
+      this.opponent = props.dataSource.members[0];
+    }
+
+    this.setState({
+      channel : props.dataSource,
+      me :this.me,
+      opponent : this.opponent,
+      lastMessageInfo : props.dataSource.lastMessage,
+      unreadCount : props.dataSource.unreadMessageCount,
+
+    });
+  }
+
+  goToChat(){
+    const channelUser = { title: this.state.opponent.nickname, me:this.state.me, opponent:this.state.opponent};
+    Actions.chatPage(channelUser);
+  }
+
+  renderUnreadCount(){
+    if (this.state.unreadCount != 0) {
+      return(
+        <View style={styles.unreadCountContainer}>
+          <Text style={styles.unreadCountText}>
+            {this.state.unreadCount}
+          </Text>
+        </View>
+      )
+    }
+    return null;
+  }
+
+  render() {
+    return (
+      <TouchableHighlight underlayColor="lightgray" onPress={this.goToChat}>
+        <View style={styles.row}>
+          <Image style={styles.photo}
+                 source={{ uri: this.state.opponent.profileUrl }}/>
+          <View style={styles.userInformation}>
+            <View style={styles.leftSection} >
+              <Text style={styles.name}>
+                {this.state.opponent.nickname}
+              </Text>
+              <Text style={styles.lastMessage}>
+                {this.state.lastMessageInfo ? this.state.lastMessageInfo.message : ''}
+              </Text>
+
+            </View>
+            <View style={styles.rightSection} >
+              <View style={styles.rightTopSection}>
+                <Text style={styles.lastTimestamp}>
+                  {
+                    this.state.lastMessageInfo ?
+                      (moment(Date.now()).startOf('day').isSame(moment(this.state.lastMessageInfo.createdAt).startOf('day')) ?
+                        moment(this.state.lastMessageInfo.createdAt)
+                          .format('LT')
+                        :
+                        moment(this.state.lastMessageInfo.createdAt)
```

Comment:
I think you can assign another variable to 'this.state.lastMessageInfo.createdAt' to avoid call moment() repeatedly